aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -421,7 +429,10 @@ Tensor sparse_compressed_to_dense(
     dense = dense.reshape({n_batch, -1, values.size(-2), values.size(-1)});
     for (auto batch : c10::irange(n_batch)) {
       Tensor batch_indices = at::_convert_indices_from_csr_to_coo(
-          crow_indices[batch], col_indices[batch], false, false);
+          compressed_indices[batch],
+          plain_indices[batch],
+          false,
+          self.layout() == kSparseBsc);
```

Comment:
It should be possible to work out some kind of trick that allows for index conversion to coo via 1/2 kernel launches rather than n_batch.